# HG changeset patch
# User Carnë Draug <carandraug+dev@gmail.com>
# Date 1350074300 -7200
#      Fri Oct 12 22:38:20 2012 +0200
# Node ID 7a0a202fedfe33e7743da277028e30ab1f66d541
# Parent  3ae8c1ee7365ecf889234eca8301500e794e71c3
iscolormap: new function for image

diff --git a/NEWS b/NEWS
--- a/NEWS
+++ b/NEWS
@@ -88,20 +88,21 @@ Summary of important user-visible change
     January 1st of the current year.  The previous default was the current day,
     month, and year.  This may produce changes in existing scripts.
 
  ** The default name of the Octave crash dump file is now called
     octave-workspace instead of octave-core.
       
  ** Other new functions added in 3.8.0:
 
-      betaincinv   erfcinv      polyeig      shrinkfaces 
-      cmpermute    findfigs     splinefit
-      cmunique     fminsearch   tetramesh
-      colorcube    lines        rgbplot     
+      betaincinv   erfcinv      lines      rgbplot
+      cmpermute    findfigs     polyeig    shrinkfaces
+      cmunique     fminsearch   splinefit
+      colorcube    iscolormap   tetramesh
+
  ** Deprecated functions.
 
     The following functions were deprecated in Octave 3.4 and have been
     removed from Octave 3.8.
                                            
       autocor    dispatch              is_global    strerror
       autocov    fstat                 krylovb      values  
       betai      gammai                perror               
diff --git a/doc/interpreter/image.txi b/doc/interpreter/image.txi
--- a/doc/interpreter/image.txi
+++ b/doc/interpreter/image.txi
@@ -112,16 +112,18 @@ A pixel in a binary image is black if it
 if it is @code{true}.
 
 An indexed image consists of an M-by-N matrix of integers
 and a C-by-3 color map.  Each integer corresponds to an
 index in the color map, and each row in the color map corresponds to
 an RGB color.  The color map must be of class @code{double} with values
 between 0 and 1.
 
+@DOCSTRING(iscolormap)
+
 @DOCSTRING(gray2ind)
 
 @DOCSTRING(ind2gray)
 
 @DOCSTRING(rgb2ind)
 
 @DOCSTRING(ind2rgb)
 
diff --git a/scripts/image/iscolormap.m b/scripts/image/iscolormap.m
new file mode 100644
--- /dev/null
+++ b/scripts/image/iscolormap.m
@@ -0,0 +1,45 @@
+## Copyright (C) 2012 Carnë Draug
+##
+## This file is part of Octave.
+##
+## Octave is free software; you can redistribute it and/or modify it
+## under the terms of the GNU General Public License as published by
+## the Free Software Foundation; either version 3 of the License, or (at
+## your option) any later version.
+##
+## Octave is distributed in the hope that it will be useful, but
+## WITHOUT ANY WARRANTY; without even the implied warranty of
+## MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+## General Public License for more details.
+##
+## You should have received a copy of the GNU General Public License
+## along with Octave; see the file COPYING.  If not, see
+## <http://www.gnu.org/licenses/>.
+
+## -*- texinfo -*-
+## @deftypefn {Function File} {} iscolormap (@var{cm})
+## Return true if @var{cm} is a colormap.
+##
+## A colormap is an @var{n} row by 3 column matrix.  The columns contain red,
+## green, and blue intensities respectively.  All entries should be between 0
+## and 1 inclusive.
+##
+## @seealso{colormap, rgbplot}
+## @end deftypefn
+
+## Author: Carnë Draug <carandraug+dev@gmail.com>
+
+function retval = iscolormap (cm)
+
+  if (nargin != 1)
+    print_usage;
+  endif
+
+  retval = (ismatrix (cm) && isreal (cm) && isnumeric (cm) &&
+            columns(cm) == 3 && ndims (cm) == 2 && isa (cm, "double") &&
+            min (cm(:)) >= 0 && max (cm(:)) <= 1);
+
+endfunction
+
+%!assert (iscolormap (jet (64)))
+%!assert (iscolormap (magic (4)), false)
diff --git a/scripts/image/module.mk b/scripts/image/module.mk
--- a/scripts/image/module.mk
+++ b/scripts/image/module.mk
@@ -13,16 +13,17 @@ image_FCN_FILES = \
   image/copper.m \
   image/flag.m \
   image/gmap40.m \
   image/gray.m \
   image/gray2ind.m \
   image/hot.m \
   image/hsv.m \
   image/hsv2rgb.m \
+  image/iscolormap.m \
   image/image.m \
   image/imagesc.m \
   image/imfinfo.m \
   image/imread.m \
   image/imshow.m \
   image/imwrite.m \
   image/ind2gray.m \
   image/ind2rgb.m \
